File: app/src/main/java/com/wizeline/daggerjava/di/AppBinder.java
package com.wizeline.daggerjava.di;

import com.wizeline.daggerjava.ui.main.DemoFragment;
import com.wizeline.daggerjava.ui.main.MainActivity;
import com.wizeline.daggerjava.ui.service.DemoService;
import com.wizeline.daggerjava.ui.service.ServiceModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
abstract class AppBinder {

    @ContributesAndroidInjector
    abstract MainActivity bindMainActivity();

    @ContributesAndroidInjector
    abstract DemoFragment bindDemoFragment();

    @ContributesAndroidInjector(modules = ServiceModule.class)
    abstract DemoService bindDemoService();
}

File: app/src/main/java/com/wizeline/dagger/di/AppBinder.kt
package com.wizeline.dagger.di

import com.wizeline.dagger.view.main.FragmentBinder
import com.wizeline.dagger.view.main.GoodActivity
import com.wizeline.dagger.view.main.GoodActivityModule
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class AppBinder {
    @ContributesAndroidInjector(modules = [GoodActivityModule::class, FragmentBinder::class])
    abstract fun whatever(): GoodActivity

    @ContributesAndroidInjector
    :Fragment
}
